src/utils/tracked-controls.js

```diff
@@ -88,19 +88,15 @@ function findMatchingController (controllers, filterIdExact, filterIdPrefix, fil
   var i;
   var matchingControllerOccurence = 0;
   var targetControllerMatch = filterControllerIndex || 0;
-  var filterIdPrefixes;
+  var filterIdRegex;
   if (filterIdPrefix && filterIdPrefix.indexOf('|') >= 0) {
-    filterIdPrefixes = filterIdPrefix.split('|');
+    filterIdRegex = new RegExp('^' + filterIdPrefix);
```

Comment:
You don't need to create a Regexp for each time we ant to find and match controllers. Can we instantiate and reuse?

Reply:
I considered that, but for example https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/compile says it's deprecated so didn't seem great.  Thoughts?

Reply:
My understanding is that because it is passed to tracked-controls, we can't simply make GAMEPAD_ID_PREFIX a RegExp (which is what both @DigiTec and I would ideally prefer) as that will not carry over properly.  As a slightly clunky workaround to prevent subsequent construction, we can probably create a global map that holds the RegExp created for a given prefix string... Is that the sort of thing you had in mind here?

Reply:
Can we make filterIdPrefix parameter to take a String or an Array an avoid using RegExps that are always hard to understand and maintain?

Reply:
The prefix needs to be passed into tracked-controls, not sure that would work as string-or-array.  The original version used split to make the array if it saw pipe delimiter, which also makes garbage, but is easier to read and maintain; would you prefer a different PR that doesn't use RegExp at all, but instead provides an alternative solution to the related issues? (i.e. simple bugfix to previous pipe delimited split)

Reply:
I would prefer to avoid RegExp if we can. When I saw `var GAMEPAD_ID_PREFIX = '((Gear VR)|(Oculus Go))'; ` it made me feel confused right off the bat :)

Reply:
regex is fine for me. it's the fastest way. probably just rename the var is good.